### PR TITLE
Fix Safari flickering and perf issues

### PR DESCRIPTION
**Pull request by pbakaus**
This PR solves numerous Safari issues:

* Special flag for Uglify to work around a let block scoping Safari 10 bug
* Refactor away Web Animations as the polyfill didn't play well with fill modes or restoring state onfinish

This should solve most flickering issues, if not all.

### CODE REVIEW

File: amp-pwa-reader/src/js/Article.js [hunk 1]
```diff
@@ -138,15 +138,16 @@ class Article {
...
-    this.container.style.opacity = 1;
-
     return new Promise(resolve => {
-      this.container.animate([
-        { opacity: 0, transform: 'translateY(' + (offset + scrollY) + 'px)' },
-        { opacity: 1, transform: 'translateY(' + scrollY + 'px)' }
-      ], { duration: this.cssVariables.animationSpeedIn, easing: 'cubic-bezier(0.23, 1, 0.32, 1)' }).onfinish = resolve;
+
+      this._transitionEndCallback = () => {
+        this.container.removeEventListener('transitionend', this._transitionEndCallback);
```
Comment (dvoytenko) [inline amp-pwa-reader/src/js/Article.js]: No longer care about `webkitTransitionEnd` event?

File: amp-pwa-reader/src/js/Article.js [hunk 2]
```diff
@@ -138,15 +138,16 @@ class Article {
...
+        resolve();
+      };
+
+      this.container.addEventListener('transitionend', this._transitionEndCallback, false);
```
Comment (dvoytenko) [inline amp-pwa-reader/src/js/Article.js]: One constant annoyance: if it so happens that destination == target, transitionend never fires :( Not sure how possible it is here. Typically I'd employ [double tap rule](https://www.youtube.com/watch?v=JmA2WYyw-_A), e.g. via timer.

File: amp-pwa-reader/src/js/Article.js [hunk 3]
```diff
@@ -155,17 +156,19 @@ class Article {
 
     // No animation if there's no card to animate from
     if (!this.card) {
-        return new Promise(resolve => resolve());
+      return new Promise(resolve => resolve());
```
Comment (dvoytenko) [inline amp-pwa-reader/src/js/Article.js]: `return Promise.resolve()`?

File: amp-pwa-reader/src/js/Article.js [hunk 4]
```diff
@@ -155,17 +156,19 @@ class Article {
...
-    let offset = (innerWidth * this.card.imageData.ratio) / 2;
-    this.container.style.transform = 'translateY(' + (offset + scrollY) + 'px)';
-
     return new Promise(resolve => {
-      return this.container.animate([
-        { opacity: 1, transform: 'translateY(' + (scrollY) + 'px)' },
-        { opacity: 0, transform: 'translateY(' + (offset + scrollY) + 'px)' }
-      ], { duration: this.cssVariables.animationSpeedOut, easing: 'cubic-bezier(0.23, 1, 0.32, 1)' }).onfinish = resolve;
+
+      this._transitionEndCallback = () => {
```
Comment (dvoytenko) [inline amp-pwa-reader/src/js/Article.js]: If you are creating and destroying this function in this one scope, is there a reason to save it as a `this` var?

File: amp-pwa-reader/src/sass/_article.scss [hunk 5]
```diff
@@ -18,6 +8,21 @@
     z-index: 5;
 }
 
+.sr-article {
+    position: fixed;
+    transform: translateY(50vh);
+    transition: transform var(--animation-speed-out) var(--animation-easing),
```
Comment (dvoytenko) [inline amp-pwa-reader/src/sass/_article.scss]: Compatibility not an issue here?

File: amp-pwa-reader/src/js/Article.js (hunk 1 above)
Comment (pbakaus) [inline amp-pwa-reader/src/js/Article.js]: Nope: http://shouldiprefix.com/#transitions :)

File: amp-pwa-reader/src/js/Article.js (hunk 2 above)
Comment (pbakaus) [inline amp-pwa-reader/src/js/Article.js]: True, but in this case it's impossible to happen, as this animation stays static for all articles.

File: amp-pwa-reader/src/sass/_article.scss (hunk 5 above)
Comment (pbakaus) [inline amp-pwa-reader/src/sass/_article.scss]: nope: http://shouldiprefix.com/#transitions

File: amp-pwa-reader/src/js/Article.js (hunk 4 above)
Comment (pbakaus) [inline amp-pwa-reader/src/js/Article.js]: good catch! fixed.

File: amp-pwa-reader/src/js/Article.js (hunk 3 above)
Comment (pbakaus) [inline amp-pwa-reader/src/js/Article.js]: fixed!

**Pull request merged**